addons/hr/models/res_users.py

```diff
@@ -1,13 +1,20 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, fields
+from odoo import api, models, fields, _
 
 
 class User(models.Model):
     _inherit = ['res.users']
 
     employee_ids = fields.One2many('hr.employee', 'user_id', string='Related employees')
+    employee_warning = fields.Boolean(string="Employee Warning", compute='_compute_employee_warning')
+
+    @api.multi
+    def _compute_employee_warning(self):
+        for user in self.filtered(lambda x: not x.employee_ids):
+            if user.has_group('hr.group_hr_user') and user.has_group('base.group_user'):
```

Comment:
Looks like `group_hr_user` implies `group_user`.
Since this message only applied to internal user, you can use the `share` field on res.users. Add `depends(groups_id)`.
Also, avoid the filtered, as it will create new prefetch, which is not good for performance.